tests/unit/test_args.py

```diff
@@ -459,3 +459,226 @@ def test_client_config_beats_config_store(self):
+
+        self.assertEqual(bins['AWS::Region'], 'ca-central-1')
+        self.assertEqual(bins['AWS::UseFIPS'], False)
+        self.assertEqual(bins['AWS::UseDualStack'], False)
+        self.assertEqual(bins['AWS::STS::UseGlobalEndpoint'], True)
+        self.assertEqual(bins['AWS::S3::UseGlobalEndpoint'], False)
+        self.assertEqual(bins['AWS::S3::Accelerate'], False)
+        self.assertEqual(bins['AWS::S3::ForcePathStyle'], False)
+        self.assertEqual(bins['AWS::S3::UseArnRegion'], True)
+        self.assertEqual(bins['AWS::S3Control::UseArnRegion'], False)
+        self.assertEqual(
+            bins['AWS::S3::DisableMultiRegionAccessPoints'], False
+        )
+        self.assertEqual(bins['SDK::Endpoint'], None)
+
+    def test_builtin_aws_region(self):
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            region_name='my-region-1',
+        )
+        self.assertEqual(bins['AWS::Region'], 'my-region-1')
+
+    def test_builtin_aws_use_fips(self):
+        # The only reason for this builtin to not have the default value
+        # (False) is that the ``use_fips_endpoint`` config variable resolves
+        # to True.
+        self.bridge._resolve_endpoint_variant_config_var = mock.Mock(
+            return_value=True
+        )
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults()
+        self.assertEqual(bins['AWS::UseFIPS'], True)
+        self.bridge._resolve_endpoint_variant_config_var.assert_called_once_with(
+            'use_fips_endpoint'
+        )
+
+    def test_builtin_aws_use_dualstack(self):
+        # The only reason for this builtin to not have the default value
+        # (False) is that the ``_resolve_use_dualstack_endpoint`` method
+        # of the ClientEndpointBridge object returns False.
+        self.bridge._resolve_use_dualstack_endpoint = mock.Mock(
+            return_value=True
+        )
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            service_name='test-service'
+        )
+        self.assertEqual(bins['AWS::UseDualStack'], True)
+        self.bridge._resolve_use_dualstack_endpoint.assert_called_once_with(
+            'test-service'
+        )
+
+    def test_builtin_aws_sts_global_endpoint(self):
+        # The only reason for this builtin to not have the default value
+        # (True) is that the ``_should_set_global_sts_endpoint`` method
+        # returns False.
+        self.args_create._should_set_global_sts_endpoint = mock.Mock(
+            return_value=False
+        )
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            region_name='my-region-2',
+        )
+        self.assertEqual(bins['AWS::STS::UseGlobalEndpoint'], False)
+        self.args_create._should_set_global_sts_endpoint.assert_called_once_with(
+            region_name='my-region-2',
+            endpoint_url=None,
+            endpoint_config=None,
+        )
```

Comment:
This feels brittle if we add another arg to the method. That's probably solved if we focus on the output for the builtin.